presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -318,49 +383,58 @@ else if (!forBroker) {
             String orderByExpressions = topNColumnOrderingMap.entrySet().stream().map(entry -> selections.get(entry.getKey()).getDefinition() + (entry.getValue().isAscending() ? "" : " DESC")).collect(Collectors.joining(", "));
             query = query + " ORDER BY " + orderByExpressions;
         }
+        query = query + limitClause;
+        return query;
+    }
+
+    /**
+     * Convert the current context to a Pinot SQL
+     */
+    public PinotQueryGenerator.GeneratedPinotQuery toSqlQuery(PinotConfig pinotConfig, ConnectorSession session)
+    {
+        int nonAggregateShortQueryLimit = PinotSessionProperties.getNonAggregateLimitForBrokerQueries(session);
+        boolean isQueryShort = hasAggregation() || limit.orElse(Integer.MAX_VALUE) < nonAggregateShortQueryLimit;
+        boolean forBroker = !PinotSessionProperties.isForbidBrokerQueries(session) && isQueryShort;
+        if (!pinotConfig.isAllowMultipleAggregations() && aggregations > 1 && !groupByColumns.isEmpty()) {
+            throw new PinotException(PINOT_QUERY_GENERATOR_FAILURE, Optional.empty(), "Multiple aggregates in the presence of group by is forbidden");
+        }
+        String expressions = selections.entrySet().stream()
+                .map(s -> s.getValue().getDefinition())
+                .collect(Collectors.joining(", "));
+        String tableName = from.orElseThrow(() -> new PinotException(PINOT_QUERY_GENERATOR_FAILURE, Optional.empty(), "Table name not encountered yet"));
+
         // Rules for limit:
         // - If its a selection query:
         //      + given limit or configured limit
         // - Else if has group by:
-        //      + ensure that only one aggregation
         //      + default limit or configured top limit
+        // - Aggregation only query limit is 1
```

Comment:
Why is this even needed ? Ideally, aggregation queries shouldn't even have a LIMIT clause in the Pinot's SQL query.

Reply:
Right. For aggregation only query, the limit is kind of ignored, unless user really do `LIMIT 0`.
I will change to remove the limit here.